File: CPP/Algorithm/Bubble_Sort.cpp
// Complexity: O(n^2)
// Iterative approach:

void bsrt(auto &v, auto n) {
    for (int i = 0; i < n - 1; ++i) {
        bool ck = true;
        for (int j = 0; j < n - i - 1; ++j) {
            if (v[j] > v[j+1]) {
                swap(v[j], v[j+1]); ck = false;
            }
        }
        if(ck) break;
    }
}

// Recursive approach:

void bsrt(auto &v, auto n) {
    if (n < 2) return;
    for (int i = 0; i < n - 1; ++i) {
        if (v[i] > v[i + 1]) swap(v[i], v[i + 1]);
    }
    bsrt(v, n - 1);
}

File: Python/Algorithm/Bubble_Sort.py
# Complexity: O(n^2)
# Iterative approach:

def bsrt():
    for i in range(n-1):
        ok = True
        for j in range(n-i-1):
            if ls[j] > ls[j+1]:
                ls[j], ls[j+1] = ls[j+1], ls[j]; ok = False
        if ok: break

# Recursive approach

def bsrt(a = None):
    if a is None: a = n
    if a <= 1: return
    for i in range(a-1):
        if ls[i] > ls[i+1]:
            ls[i], ls[i+1] = ls[i+1], ls[i]  # Swapping elements
    bsrt(a-1)